src/components/grid/grid.ts

```diff
@@ -303,6 +303,40 @@ export class Grid extends Component<Props, SpreadsheetChildEnv> {
       const position = this.env.model.getters.getActivePosition();
       this.env.model.selection.selectZone({ cell: position, zone: newZone });
     },
+    "CTRL+D": async () => {
+      if (this.env.model.getters.getSelectedZones().length === 1) {
+        const zone = this.env.model.getters.getSelectedZone();
+        const target =
+          zone.top !== zone.bottom
+            ? {
+                ...zone,
+                bottom: zone.top,
+              }
+            : {
+                ...zone,
+                top: zone.top - 1,
+                bottom: zone.top - 1,
+              };
```

Comment:
```suggestion
        const singleRowSelected  = zone.top !== zone.bottom
        const target ={
                ...zone,
                bottom: singleRowSelected ? zone.top : zone.top - 1,
                top: singleRowSelected ? zone.top : zone.top - 1,
              }
```
shorter and way easier to read IMO

same for the other shortcut